From 7d3109f7065a11fc81fe5698ec4274dc840cfc60 Mon Sep 17 00:00:00 2001
From: Cezary Rojewski <cezary.rojewski@intel.com>
Date: Sun, 27 Jan 2019 20:40:01 +0100
Subject: [PATCH] ASoC: Intel: Skylake: Correct skl_base_cfg declaration.

As per FW Interface Modules Configuration, init instance IPC request
requires base initial module configuration. This configuration structure
is made of: cpc, ibs, obs, is_pages and audio_fmt.

Current implementation incorrectly uses cps instead of cpc, causing
issues with initialization and scheduling of DP modules.

Change-Id: I5426262df7a0691d72db94d517d488e8281a2199
Signed-off-by: Cezary Rojewski <cezary.rojewski@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/35416
Tested-by: Bozek, DominikX <dominikx.bozek@intel.com>
Reviewed-by: Slawinski, AmadeuszX <amadeuszx.slawinski@intel.com>

ASoC: Intel: Skylake: Fixed compilation failure for CNL

Due to changes in skl_base_fmt from ASoC: Intel: Skylake: Correct skl_base_cfg declaration
compilation issue is faced since there is no cps member, updating the cps values to cpc*1000

Change-Id: Ida7e1e8850d62b0152110e9c550210e6a673c847
Signed-off-by: Gaurav Vohra <gauravx.vohra@intel.com>
---
 sound/soc/intel/skylake/skl-debug.c    |  8 +++++---
 sound/soc/intel/skylake/skl-messages.c |  6 +++---
 sound/soc/intel/skylake/skl-topology.c | 14 ++++++--------
 sound/soc/intel/skylake/skl-topology.h |  9 ++++-----
 4 files changed, 18 insertions(+), 19 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-debug.c b/sound/soc/intel/skylake/skl-debug.c
index e5acb86..67d8aa8 100644
--- a/sound/soc/intel/skylake/skl-debug.c
+++ b/sound/soc/intel/skylake/skl-debug.c
@@ -315,6 +315,8 @@ static ssize_t module_read(struct file *file, char __user *user_buf,
 			   size_t count, loff_t *ppos)
 {
 	struct skl_module_cfg *mconfig = file->private_data;
+	struct skl_module *module = mconfig->module;
+	struct skl_module_res *res = &module->resources[mconfig->res_idx];
 	char *buf;
 	ssize_t ret;
 
@@ -327,9 +329,9 @@ static ssize_t module_read(struct file *file, char __user *user_buf,
 			mconfig->id.module_id, mconfig->id.instance_id,
 			mconfig->id.pvt_id);
 
-	ret += scnprintf(buf + ret, MOD_BUF - ret,
-			"Resources:\n\tMCPS %#x\n\tIBS %#x\n\tOBS %#x\t\n",
-			mconfig->mcps, mconfig->ibs, mconfig->obs);
+	ret += snprintf(buf + ret, MOD_BUF - ret,
+			"Resources:\n\tCPC %#x\n\tIBS %#x\n\tOBS %#x\t\n",
+			res->cpc, res->ibs, res->obs);
 
 	ret += scnprintf(buf + ret, MOD_BUF - ret,
 			"Module data:\n\tCore %d\n\tIn queue %d\n\t"
diff --git a/sound/soc/intel/skylake/skl-messages.c b/sound/soc/intel/skylake/skl-messages.c
index bbd696f..937cc09 100644
--- a/sound/soc/intel/skylake/skl-messages.c
+++ b/sound/soc/intel/skylake/skl-messages.c
@@ -467,7 +467,7 @@ static int cnl_sdw_bra_pipe_cfg_pb(struct skl_sst *ctx,
 		goto error;
 	}
 
-	host_cpr_cfg->module->resources[0].cps = 100000;
+	host_cpr_cfg->module->resources[0].cpc = 100000000;
 	host_cpr_cfg->module->resources[0].is_pages = 0;
 	host_cpr_cfg->module->resources[0].ibs = 384;
 	host_cpr_cfg->module->resources[0].obs = 384;
@@ -729,7 +729,7 @@ static int cnl_sdw_bra_pipe_cfg_cp(struct skl_sst *ctx,
 		goto error;
 	}
 
-	link_cpr_cfg->module->resources[0].cps = 100000;
+	link_cpr_cfg->module->resources[0].cpc = 100000000;
 	link_cpr_cfg->module->resources[0].is_pages = 0;
 	link_cpr_cfg->module->resources[0].ibs = 1152;
 	link_cpr_cfg->module->resources[0].obs = 1152;
@@ -1437,7 +1437,7 @@ static void skl_set_base_module_format(struct skl_sst *ctx,
 
 	base_cfg->audio_fmt.interleaving = format->interleaving_style;
 
-	base_cfg->cps = res->cps;
+	base_cfg->cpc = res->cpc;
 	base_cfg->ibs = res->ibs;
 	base_cfg->obs = res->obs;
 	base_cfg->is_pages = res->is_pages;
diff --git a/sound/soc/intel/skylake/skl-topology.c b/sound/soc/intel/skylake/skl-topology.c
index 5f62a04..b7e18ef 100644
--- a/sound/soc/intel/skylake/skl-topology.c
+++ b/sound/soc/intel/skylake/skl-topology.c
@@ -3571,9 +3571,6 @@ static int skl_tplg_fill_res_tkn(struct device *dev,
 		return -EINVAL;
 
 	switch (tkn_elem->token) {
-	case SKL_TKN_MM_U32_CPS:
-		res->cps = tkn_elem->value;
-		break;
 
 	case SKL_TKN_MM_U32_DMA_SIZE:
 		res->dma_buffer_size = tkn_elem->value;
@@ -3595,10 +3592,6 @@ static int skl_tplg_fill_res_tkn(struct device *dev,
 		res->ibs = tkn_elem->value;
 		break;
 
-	case SKL_TKN_U32_MAX_MCPS:
-		res->cps = tkn_elem->value;
-		break;
-
 	case SKL_TKN_MM_U32_RES_PIN_ID:
 	case SKL_TKN_MM_U32_PIN_BUF:
 		ret = skl_tplg_manifest_pin_res_tkn(dev, tkn_elem, res,
@@ -3607,6 +3600,11 @@ static int skl_tplg_fill_res_tkn(struct device *dev,
 			return ret;
 		break;
 
+	case SKL_TKN_MM_U32_CPS:
+	case SKL_TKN_U32_MAX_MCPS:
+		/* ignore unused tokens */
+		break;
+
 	default:
 		dev_err(dev, "Not a res type token: %d", tkn_elem->token);
 		return -EINVAL;
@@ -4105,7 +4103,7 @@ static int skl_tplg_get_pvt_data_v4(struct snd_soc_tplg_dapm_widget *tplg_w,
 		return ret;
 	mconfig->id.module_id = -1;
 	mconfig->id.instance_id = dfw->instance_id;
-	mconfig->module->resources[0].cps = dfw->max_mcps;
+	mconfig->module->resources[0].cpc = dfw->max_mcps/1000;
 	mconfig->module->resources[0].ibs = dfw->ibs;
 	mconfig->module->resources[0].obs = dfw->obs;
 	mconfig->core_id = dfw->core_id;
diff --git a/sound/soc/intel/skylake/skl-topology.h b/sound/soc/intel/skylake/skl-topology.h
index b78ac11..faf045a 100644
--- a/sound/soc/intel/skylake/skl-topology.h
+++ b/sound/soc/intel/skylake/skl-topology.h
@@ -122,7 +122,7 @@ struct skl_audio_data_format {
 } __packed;
 
 struct skl_base_cfg {
-	u32 cps;
+	u32 cpc;
 	u32 ibs;
 	u32 obs;
 	u32 is_pages;
@@ -388,7 +388,6 @@ struct skl_module_pin_resources {
 struct skl_module_res {
 	u8 id;
 	u32 is_pages;
-	u32 cps;
 	u32 ibs;
 	u32 obs;
 	u32 dma_buffer_size;
@@ -461,9 +460,9 @@ struct skl_module_cfg {
 	u8 out_queue_mask;
 	u8 in_queue;
 	u8 out_queue;
-	u32 mcps;
-	u32 ibs;
-	u32 obs;
+        u32 mcps;
+        u32 ibs;
+        u32 obs;
 	u8 is_loadable;
 	u8 core_id;
 	u8 dev_type;
-- 
1.9.1

